gensim/models/normmodel.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2012 Radim Rehurek <radimrehurek@seznam.cz>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+import logging
+
+from gensim import interfaces, matutils
+
+logger = logging.getLogger(__name__)
+
+
+class Normmodel(interfaces.TransformationABC):
+    """
+    Objects of this class realize the explicit normalization of
+    vectors. Supported norms are l1' and 'l2' with 'l2' being
+    default.
+
+    The main methods are:
+
+    1. Constructor which normalizes the terms in the given corpus document-wise.
+    2. The normalize() method which normalizes a simple count representation.
+
+    >>> norm_l2 = Normmodel(corpus)
+    >>> print(norm_l2[some_doc])
+    >>> norm_l2.save('/tmp/foo.tfidf_model')
+
+    Model persistency is achieved via its load/save methods
+    """
+    def __init__(self, corpus=None, norm='l2'):
+        """
+        Compute the 'l1' or 'l2' normalization by normalizing separately
+        for each doc in a corpus.
+        Formula for 'l1' norm for term 'i' in document 'j' in a corpus of 'D' documents is::
+
+          norml1_{i, j} = (i / sum(absolute(values in j)))
+
+        Formula for 'l2' norm for term 'i' in document 'j' in a corpus of 'D' documents is::
+
+          norml2_{i, j} = (i / sqrt(sum(square(values in j))))
+        """
+        self.norm = matutils.isnorm(norm)
+        if corpus is not None:
+            self.calc_norm(corpus)
+        else:
+            pass
+
+    def __str__(self):
+        return "Normmodel(num_docs=%s, num_nnz=%s)" % (self.num_docs, self.num_nnz)
+
+    def calc_norm(self, corpus):
+        """
+        Calculates the norm by calling matutils.unitvec with the norm parameter.
+        """
+        logger.info("Normalizing...")
+        norms = {}
+        numnnz, docno = 0, -1
+        for docno, bow in enumerate(corpus):
+            numnnz += len(bow)
+            norms[docno] = matutils.unitvec(bow, self.norm)
```

Comment:
Sorry I didn't understand this. The norms dictionary itself is being constructed by iterating through all the docs in the corpus in this loop. Did you mean rather than using `enumerate(corpus)` I can simply iterate through the corpus and use `len(norms)` for indexing? While writing this, I personally felt that explicitly mentioning `docno` would enhance the readability of the code.....